collect_app/src/main/java/org/odk/collect/android/adapters/FileArrayAdapter.java

```diff
@@ -31,113 +31,84 @@
 import org.odk.collect.android.logic.DriveListItem;
 
 import java.text.SimpleDateFormat;
-import java.util.Collections;
-import java.util.Comparator;
 import java.util.Date;
 import java.util.List;
 import java.util.Locale;
 
-public class FileArrayAdapter extends RecyclerView.Adapter<FileArrayAdapter.ViewHolder> {
+import android.support.annotation.NonNull;
+
+public class FileArrayAdapter extends ArrayAdapter<DriveListItem> {
 
     private Context context;
     private List<DriveListItem> items;
 
-    public interface OnItemClickListener {
-        void onItemClick(View v, DriveListItem item);
-    }
-
-    private final OnItemClickListener listener;
 
-    public FileArrayAdapter(Context context, List<DriveListItem> objects, OnItemClickListener listener) {
+    public FileArrayAdapter(Context context, List<DriveListItem> filteredDriveList) {
+        super(context, R.layout.two_item_image, filteredDriveList);
         this.context = context;
```

Comment:
This global variable `context` can be removed and `getContext()` can be used in its place wherever required